impeller/display_list/display_list_image_impeller_raster.cc

```diff
@@ -0,0 +1,167 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/display_list/display_list_image_impeller_raster.h"
+
+#include "flutter/fml/trace_event.h"
+#include "flutter/impeller/core/allocator.h"
+#include "flutter/impeller/core/texture.h"
+#include "flutter/impeller/renderer/command_buffer.h"
+#include "impeller/aiks/aiks_context.h"
+#include "impeller/entity/contents/filters/filter_contents.h"
+
+namespace impeller {
+
+static std::optional<impeller::PixelFormat> ToPixelFormat(SkColorType type) {
+  switch (type) {
+    case kRGBA_8888_SkColorType:
+      return impeller::PixelFormat::kR8G8B8A8UNormInt;
+    case kRGBA_F16_SkColorType:
+      return impeller::PixelFormat::kR16G16B16A16Float;
+    case kBGR_101010x_XR_SkColorType:
+      return impeller::PixelFormat::kB10G10R10XR;
+    default:
+      return std::nullopt;
+  }
+  return std::nullopt;
+}
```

Comment:
This is copied from another file, can we share this logic?

Reply:
Yeah, I meant to make this shared.